### PR TITLE
Fix log rotation description in the logging doc

### PR DESCRIPTION
**Pull request by crassirostris**
To address https://github.com/kubernetes/kubernetes/pull/40634#issuecomment-303478050

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3918)

### CODE REVIEW

File: docs/concepts/cluster-administration/logging.md [hunk 1]
```diff
@@ -80,14 +93,16 @@ that do not run in a container. For example:
...
-bypassing the default logging mechanism. They use the [glog](https://godoc.org/github.com/golang/glog)
+bypassing the default logging mechanism. They use the [glog][glog]
 logging library. You can find the conventions for logging severity for those
 components in the [development docs on logging](https://github.com/kubernetes/community/blob/master/contributors/devel/logging.md).
 
 Similarly to the container logs, system component logs in the `/var/log`
-directory are rotated daily and based on the log size. However,
-system component logs have a higher size retention: by default,
-they can store up to 100MB.
+directory should be rotated. In Kubernetes clusters brough up by
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: "brough" -> "brought"

File: docs/concepts/cluster-administration/logging.md [hunk 2]
```diff
@@ -80,14 +93,16 @@ that do not run in a container. For example:
...
+`kube-up.sh` script, those logs are configured to be rotated by
+`logrotate` tool daily or once the size exceeds 100MB.
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "the": "...**the** `logrotate` tool..."

File: docs/concepts/cluster-administration/logging.md [hunk 3]
```diff
@@ -80,14 +93,16 @@ that do not run in a container. For example:
...
+bypassing the default logging mechanism. They use the [glog][glog]
 logging library. You can find the conventions for logging severity for those
 components in the [development docs on logging](https://github.com/kubernetes/community/blob/master/contributors/devel/logging.md).
 
 Similarly to the container logs, system component logs in the `/var/log`
-directory are rotated daily and based on the log size. However,
-system component logs have a higher size retention: by default,
-they can store up to 100MB.
+directory should be rotated. In Kubernetes clusters brough up by
+`kube-up.sh` script, those logs are configured to be rotated by
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "the": "...**the** `kube-up.sh` script..."

File: docs/concepts/cluster-administration/logging.md [hunk 4]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
-
-The Kubernetes logging configuration differs depending on the node type. For example, you can find detailed information for GCI in the corresponding [configure helper](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/cluster/gce/gci/configure-helper.sh#L96).
-
-When you run [`kubectl logs`](/docs/user-guide/kubectl/v1.6/#logs), as in the basic logging example, the kubelet on the node handles the request and reads directly from the log file, returning the contents in the response. Note that `kubectl logs` **only returns the last rotation**; you must manually extract prior rotations, if desired and cluster-level logging is not enabled.
+An important consideration in node-level logging is implementing log rotation,
+so that logs don't consume all available storage on the node. Kubernetes
+up to 1.7 is not responsible for rotating logs, deployment tool should set it
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: Suggested edit: "Kubernetes versions less than 1.7 are not responsible for rotating logs, so a deployment tool should be set up to address that."

File: docs/concepts/cluster-administration/logging.md [hunk 5]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
+up. For example, in Kubernetes clusters, deployed by `kube-up.sh` script,
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "the": "...**the** kube-up.sh script..."

File: docs/concepts/cluster-administration/logging.md [hunk 6]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
+there is [`logrotate`](http://www.linuxcommand.org/man_pages/logrotate8.html)
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "a": "...there is **a** [`logrotate`](http://www.linuxcommand.org/man_pages/logrotate8.html) tool..."

File: docs/concepts/cluster-administration/logging.md [hunk 7]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
+tool configured to perform log rotations daily or once application's log
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "an": "...or once **an** application's log..."

File: docs/concepts/cluster-administration/logging.md [hunk 8]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
+file grows beyond 10MB. You can also set up container runtime to rotate logs
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "a": "You can also set up **a** container runtime..."

File: docs/concepts/cluster-administration/logging.md [hunk 9]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
+automatically, e.g. by using Docker's `log-opt` or by sending logs to the
+system that does this for you, e.g. journald. As an example, you can find
+detailed information about how `kube-up.sh` sets up logging in the
+corresponding [script][cosConfigureHelper].
+
+When you run [`kubectl logs`](/docs/user-guide/kubectl/v1.6/#logs) as in
+the basic logging example, the kubelet on the node handles the request and
+reads directly from the log file, returning the contents in the response.
+**Note:** if some external system has manipulated the log file, e.g.
+logrotate performed rotation, only the content of the original log file
```
Comment (chenopis) [inline docs/concepts/cluster-administration/logging.md]: add "the": "...logrotate performed **the** rotation..."

**Comment by crassirostris** [comment]
@chenopis I swear when I'm creating the PR the checkbox "Allow edits from maintainers" is checked, but after PR is created, it's unchecked. That's what I was missing, sorry

File: docs/concepts/cluster-administration/logging.md (hunk 4 above)
Comment (fgrzadkowski) [inline docs/concepts/cluster-administration/logging.md]: Do we want to refer to a specific kubelet version? We'll have to update it every release.

File: docs/concepts/cluster-administration/logging.md [hunk 10]
```diff
@@ -61,13 +61,26 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
-When you run [`kubectl logs`](/docs/user-guide/kubectl/v1.6/#logs), as in the basic logging example, the kubelet on the node handles the request and reads directly from the log file, returning the contents in the response. Note that `kubectl logs` **only returns the last rotation**; you must manually extract prior rotations, if desired and cluster-level logging is not enabled.
+An important consideration in node-level logging is implementing log rotation,
+so that logs don't consume all available storage on the node. Kubernetes
+up to 1.7 is not responsible for rotating logs, deployment tool should set it
+up. For example, in Kubernetes clusters, deployed by `kube-up.sh` script,
+there is [`logrotate`](http://www.linuxcommand.org/man_pages/logrotate8.html)
+tool configured to perform log rotations daily or once application's log
+file grows beyond 10MB. You can also set up container runtime to rotate logs
+automatically, e.g. by using Docker's `log-opt` or by sending logs to the
+system that does this for you, e.g. journald. As an example, you can find
```
Comment (fgrzadkowski) [inline docs/concepts/cluster-administration/logging.md]: AFAIU `journald` doesn't work with `kubectl logs` or `/logs` http handler. I wouldn't recommend it in our docs.

File: docs/concepts/cluster-administration/logging.md (hunk 9 above)
Comment (fgrzadkowski) [inline docs/concepts/cluster-administration/logging.md]: What do you mean by "original"? I think it's unclear.

File: docs/concepts/cluster-administration/logging.md (hunk 4 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: It's still not responsible and it's unknown when/whether it'll be responsible

I'll try to make it clearer

File: docs/concepts/cluster-administration/logging.md (hunk 5 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 6 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 7 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 8 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 9 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 1 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 3 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 2 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 10 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

File: docs/concepts/cluster-administration/logging.md (hunk 9 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: ACK

**Comment by crassirostris** [comment]
@chenopis @fgrzadkowski Addressed comments, PTAL

File: docs/concepts/cluster-administration/logging.md [hunk 11]
```diff
@@ -61,13 +61,28 @@ Everything a containerized application writes to `stdout` and `stderr` is handle
...
-When you run [`kubectl logs`](/docs/user-guide/kubectl/v1.6/#logs), as in the basic logging example, the kubelet on the node handles the request and reads directly from the log file, returning the contents in the response. Note that `kubectl logs` **only returns the last rotation**; you must manually extract prior rotations, if desired and cluster-level logging is not enabled.
+An important consideration in node-level logging is implementing log rotation,
+so that logs don't consume all available storage on the node. Kubernetes
+in version less than or equal to 1.7 is not responsible for rotating logs,
+but rather a deployment tool should set up a solution to address that.
+For example, in Kubernetes clusters, deployed by the `kube-up.sh` script,
+there is a [`logrotate`](http://www.linuxcommand.org/man_pages/logrotate8.html)
+tool configured to perform log rotations daily or once an application's log
+file grows beyond 10MB. You can also set up a container runtime to rotate logs
+automatically, e.g. by using Docker's `log-opt`. As an example, you can find
```
Comment (fgrzadkowski) [inline docs/concepts/cluster-administration/logging.md]: Aren't we using Docker to rotate logs? I though we now use 'logrotate' only for logs from system components.

File: docs/concepts/cluster-administration/logging.md (hunk 4 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: > Do we want to refer to a specific kubelet version? We'll have to update it every release.
 
When I was addressing this part, I forgot to refresh the page and this comment wasn't visible

I agree, it's better to remove the mention of the version altogether

File: docs/concepts/cluster-administration/logging.md (hunk 11 above)
Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: In COS on GCP, yes. Agree, considering also the link below, this part may be confusing. I'll address it, thanks

Comment (crassirostris) [inline docs/concepts/cluster-administration/logging.md]: Done

**Comment by chenopis** [comment]
@fgrzadkowski or @piosz does this have tech lgtm?

**Comment by fgrzadkowski** [comment]
@chenopis Now it does :)

**Pull request merged**